src/test/scala/org/broadinstitute/dsde/firecloud/service/ExportEntitiesByTypeServiceSpec.scala

```diff
@@ -1,98 +1,248 @@
+        }
+      }
+    }
+
+    "when calling POST on exporting a valid collection type" - {
+      "OK response is returned" in {
+        Post(validCookieFireCloudEntitiesSampleSetTSVPath, FormData(Seq("FCtoken"->"token"))) ~> sealRoute(cookieAuthedRoutes) ~> check {
+          handled should be(true)
+          status should be(OK)
+          entity shouldNot be(empty)
+          headers.contains(HttpHeaders.Connection("Keep-Alive")) should be(true)
+          headers.contains(HttpHeaders.`Content-Disposition`.apply("attachment", Map("filename" -> "sample_set.zip"))) should be(true)
+        }
+      }
+    }
+
+    "when calling POST on exporting a valid entity type" - {
+      "OK response is returned" in {
+        Post(validCookieFireCloudEntitiesSampleTSVPath, FormData(Seq("FCtoken"->"token"))) ~> sealRoute(cookieAuthedRoutes) ~> check {
+          handled should be(true)
+          status should be(OK)
+          entity shouldNot be(empty)
+          headers.contains(HttpHeaders.Connection("Keep-Alive")) should be(true)
+          headers.contains(HttpHeaders.`Content-Disposition`.apply("attachment", Map("filename" -> "sample.txt"))) should be(true)
+        }
+      }
+    }
+
+    "when calling POST on exporting an invalid collection type" - {
+      "NotFound response is returned" in {
+        Post(invalidCookieFireCloudEntitiesParticipantSetTSVPath, FormData(Seq("FCtoken"->"token"))) ~> sealRoute(cookieAuthedRoutes) ~> check {
+          handled should be(true)
+          status should be(NotFound)
+        }
+      }
+    }
+
+    "when calling POST on exporting an invalid entity type" - {
+      "NotFound response is returned" in {
+        Post(invalidCookieFireCloudEntitiesSampleTSVPath, FormData(Seq("FCtoken"->"token"))) ~> sealRoute(cookieAuthedRoutes) ~> check {
+          handled should be(true)
+          status should be(NotFound)
+          errorReportCheck("Rawls", NotFound)
+        }
+      }
+    }
+
+    "when calling GET, PUT, PATCH, DELETE on export path" - {
+      "MethodNotAllowed response is returned" in {
+        List(HttpMethods.GET, HttpMethods.PUT, HttpMethods.DELETE, HttpMethods.PATCH) map { method =>
+          new RequestBuilder(method)(invalidCookieFireCloudEntitiesParticipantSetTSVPath) ~> sealRoute(cookieAuthedRoutes) ~> check {
+            handled should be(true)
+            status shouldNot equal(MethodNotAllowed)
+          }
+        }
+      }
+    }
+
+  }
+
+  private def validateLineCount(chunks: List[MessageChunk], count: Int): Unit = {
+    val lineCount = chunks.map(c => scala.io.Source.fromString(c.data.asString).getLines().size).sum
+    lineCount should equal(count)
+  }
+
+  private def validateProps(entity: HttpEntity): Unit = {
+    val entityHeaderString = entity.asString
+    filterProps.map { h => entityHeaderString.contains(h) should be(true) }
+    missingProps.map { h => entityHeaderString.contains(h) should be(false) }
```

Comment:
Pretty sure my question is resolved. It's true that once a response is started via the chunking process, we no longer have control over the response code. What I chose to do, after we start streaming, is to put any error that we get from either Rawls or Orchestration into the output. That way, there is at least some indication for the user that something went wrong. The tests we have check for this too.